batch/batch/batch.py

```diff
@@ -640,6 +667,17 @@ def to_dict(self):
     abort(404)
 
 
+async def _get_pod_status(batch_id, job_id, user):
+    job = await Job.from_db(batch_id, job_id, user)
+    if not job:
+        abort(404)
+
+    pod_statuses = await job._read_pod_statuses()
```

Comment:
We should get the pod statuses when we get the job the first time, why make a second trip?

Reply:
I think I addressed this. It still requires an async call to get the pod status for the current task but i use the already fetched ones for previous tasks.